### PR TITLE
Update main.yml

### PR DESCRIPTION
**Pull request by priitliivak**
For CentOS 7 rpm install apparently does not create /usr/bin/keytool and /usr/bin/javadoc links. This leads to failure if playbook is run repeatedly.

---

This change is [Reviewable](https://reviewable.io/reviews/ansiblebit/oracle-java/17)

### CODE REVIEW

File: tasks/redhat/main.yml [hunk 1]
```diff
@@ -23,6 +23,12 @@
   tags:
     - installation
 
+- name: ensure keytool and javadoc links present
+  file: src={{ item.src }} dest="/usr/bin/{{ item.dest }}" state=link
+  with_items:
+    - { src: "{{ oracle_java_home }}/jre/bin/keytool", dest: 'keytool' }
+    - { src: "{{ oracle_java_home }}/bin", dest: 'javadoc' }
+
```
Comment (steenzout) [inline tasks/redhat/main.yml:31]: probably you'll need `sudo: yes` here.

**Comment by steenzout** [comment]
@priitliivak I'm adding some tests to replicate the problem.

File: tasks/redhat/main.yml (hunk 1 above)
Comment (steenzout) [inline tasks/redhat/main.yml:31]: you also need a `when: oracle_java_set_as_default`.

**Comment by steenzout** [comment]
@priitliivak after I run this the role as it is this is what I get:

```
[vagrant@localhost ~]$ ls -lha /usr/bin/javadoc 
lrwxrwxrwx. 1 root root 25 Mar 21 15:52 /usr/bin/javadoc -> /etc/alternatives/javadoc
[vagrant@localhost ~]$ ls -lha /usr/bin/keytool 
lrwxrwxrwx. 1 root root 25 Mar 21 15:52 /usr/bin/keytool -> /etc/alternatives/keytool
```

this patch will break this.
I think the idea is to let `alternatives` determine which Java version to point to.

**Comment by steenzout** [comment]
here is the code that handles the Java version switch using `alternatives`:

```
- name: set Java version as default
  alternatives:
    name="{{ item.exe }}"
    link="/usr/bin/{{ item.exe }}"
    path="{{ item.path }}/{{ item.exe }}"
  when: oracle_java_set_as_default
  with_items:
    - { path: "{{ oracle_java_home }}/jre/bin", exe: 'java' }
    - { path: "{{ oracle_java_home }}/jre/bin", exe: 'keytool' }
    - { path: "{{ oracle_java_home }}/bin", exe: 'javac' }
    - { path: "{{ oracle_java_home }}/bin", exe: 'javadoc' }
  sudo: yes
  when: oracle_java_task_rpm_download|changed or (oracle_java_installed and oracle_java_version_installed != oracle_java_version_string)
  register: oracle_java_task_set_default
```

the problem might be the double `when` parameter... fixing that quickly.

**Comment by steenzout** [comment]
v4.9.8 tag has the fix for the double `when` parameter which should fix this problem.

let me know if you need more help.

**Comment by priitliivak** [comment]
Thanks, will try it out. Using it for my personal project so not constantly
working on it. Will try out in the end of the week.

2016-03-21 23:37 GMT+02:00 Pedro Salgado notifications@github.com:

> v4.9.8 tag has the fix for the double when parameter which should fix
> this problem.
> 
> let me know if you need more help.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> https://github.com/ansiblebit/oracle-java/pull/17#issuecomment-199497979

## 

Priit Liivak
GSM: +372 55550418

**Comment by steenzout** [comment]
closing ticket.

**Pull request closed without merging**